### PR TITLE
Ensure the save directory exists before using it

### LINKED ISSUE #186: First frame not stored if directory has to be created
If the frame export location is set to a directory that no longer exists and a frame is captured, Boats Animator proceeds to recreate the directory and save captured frames to there. However when this occurs the first frame captured is not saved to this directory.

### PR DESCRIPTION
**Pull request by le717**
Fixes #186. Also corrects the indentation of a few small places.

### CODE REVIEW

File: app/js/main.js [hunk 1]
```diff
@@ -139,14 +140,17 @@ function closeAnimator() {
 
 function startup() {
   "use strict";
-  // There is no set save directory, set one
-  if (!_getSaveDirectory()) {
+
+  let saveDirectory = _getSaveDirectory();
+
+  // There is no set save directory or the directory does not exist
+  if (!_checkSaveDirectory(saveDirectory)) {
```
Comment (BoatsAreRockable) [inline app/js/main.js]: If this check fails some method of blocking the capturing of frames needs to occur

File: app/js/main.js [hunk 2]
```diff
@@ -754,27 +762,27 @@ function _displaySaveDirectory(dir) {
...
-  return localStorage.getItem("default_directory");
+  return localStorage.getItem("save-directory");
 }
 
 /**
- * Create the frame save directory.
+ * Create the app save directory.
  *
- * @param {String} - The directory to create.
+ * @param {String} dir - The directory to create.
```
Comment (BoatsAreRockable) [inline app/js/main.js]: Good spot!

**Comment by le717** [comment]
I would approach it a different way. If no directory is set, the button should be `disabled=true` (or something to that effect), then when a directory is finally set, the button is reenabled. Also ensure the blocking affects the shortcuts. I'd rather disable functionality than add checks like that in places they don't really fit (like `takePicture()`).

File: app/js/main.js (hunk 2 above)
Comment (le717) [inline app/js/main.js]: Heh, yea, I flubbed this last PR... oops

**Comment by BoatsAreRockable** [comment]
Hi @le717, your approach to fixing this sounds good! 😄 Would you like to submit a fix? Or shall I do so?

**Comment by le717** [comment]
I've hopefully fixed everything. Take another look and let me know. :)

**Comment by BoatsAreRockable** [comment]
This does appear to now be fixed. Many thanks! 😄

**Pull request merged**